### ISSUE TITLE
class_id support for PrecisionAtRecall, RecallAtPrecision, and related metrics

### ISSUE THREAD

**Issue by dwyatte** [tensorflow/tensorflow#46017]
**System information**
- TensorFlow version (you are using): 2.4.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
`tf.keras.metrics.Precision` and `tf.keras.metrics.Recall` both include an optional `class_id` keyword argument to compute metrics in a multi-class setting for a given `class_id`. It would be nice if other metrics shared this feature including `PrecisionAtRecall` and `RecallAtPrecision` (could probably easily be added to any metric that uses `metrics_utils.update_confusion_matrix_variables`).

**Will this change the current api? How?** No, `class_id` is an optional keyword argument.

**Who will benefit with this feature?** Users wanting to monitor class-specific metrics in a multi-class setting.

**Comment by dwyatte** [comment]
PR to add support for `class_id` on all subclasses of `SensitivitySpecificityBase`: https://github.com/tensorflow/tensorflow/pull/46099

**Comment by dwyatte** [comment]
Closed by https://github.com/tensorflow/tensorflow/pull/46099
